### PR TITLE
feat: support internal metaName_TARGET env vars

### PR DESCRIPTION
**Pull request by zllkjc**
# PR Details

In some cases, the code needs to distinguish between SSR and CSR in order to remove environment-related code.

```tsx
const App = () => {
  if (process.env.MODERN_TARGET === 'browser') {
    const safeGet = require('lodash.get');
    const o = { a: 'x' };
    console.log(safeGet(o, 'a'));
    // or some sdk only run in browser.
  };
};

export default App;
```

Unlike `context.isBrowser`, this env var removes useless code after production bundle.

`context.isBrowser` can be used in most cases to differentiate logic at runtime.

But in some cases, such as files with side effects, or logic outside the component, you need to do it through this environment variables：

```ts
logDevice(window.innerHeight);
```

We also recommend using environment variables instead of the runtime, as this will make the application package size

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/runtime/plugin-runtime/src/ssr/cli/index.ts [hunk 1]
```diff
@@ -76,6 +76,26 @@ export default (): CliPlugin<AppTools> => ({
                     { filename: LOADABLE_STATS_FILE },
                   ]);
               }
+
+              // add environment variables to determine the node/browser
+              const prefix = `${
+                appContext.metaName.split(/[-_]/)[0]
+              }_`.toUpperCase();
+              const jupiterVars = {
```
Comment (chenjiahan) [inline packages/runtime/plugin-runtime/src/ssr/cli/index.ts:84]: jupiterVars -> modernVars

File: packages/toolkit/main-doc/zh/guides/basic-features/env-vars.md [hunk 2]
```diff
@@ -23,6 +23,36 @@ Modern.js 提供了对环境变量的支持，包含内置的环境变量和自
...
+    console.log(window.innerHeight);
+  };
+};
+```
+
+打包后可以看到 SSR 产物和 CSR 产物如下：
+
+```js title="dist/bundles/main.js"
+function App() {
+  if (false) {}
```
Comment (xuchaobei) [inline packages/toolkit/main-doc/zh/guides/basic-features/env-vars.md:42]: if 语句会被移除掉吧？

**Review by xuchaobei — CHANGES_REQUESTED** [review]

File: packages/runtime/plugin-runtime/src/ssr/cli/index.ts (hunk 1 above)
Comment (zllkjc) [reply packages/runtime/plugin-runtime/src/ssr/cli/index.ts]: done

File: packages/toolkit/main-doc/zh/guides/basic-features/env-vars.md (hunk 2 above)
Comment (zllkjc) [reply packages/toolkit/main-doc/zh/guides/basic-features/env-vars.md:42]: 嗯嗯，我补充了一个 note，这里是开发环境的打包产物。

**Review by xuchaobei — APPROVED** [review]

**Pull request merged**